simphony_ui/ui.py

```
@@ -186,12 +216,15 @@ def _run_calc_threaded(self):
         """ Function which will run the calculation. This function
         is only run by the secondary thread
         """
-        return run_calc(
-            self.global_settings,
-            self.openfoam_settings,
-            self.liggghts_settings,
-            self.update_progress_bar
-        )
+        try:
+            return run_calc(
+                self.global_settings,
+                self.openfoam_settings,
+                self.liggghts_settings,
+                self.update_progress_bar
+            )
+        except Exception:
```

This is a bit dangerous, because you swallow any exception and leave no trace to understand what happened. You are in the secondary thread here, so you don't have a lot of freedom, but what you can do is to use the logging module to log the exception. check log.exception() (You will have to create a logger with getLogger(), use the name of the class)

It's a quite common pattern that you have a `log` trait in Application classes.

---

Take care with setting up a logger here: library code should be careful to do ```logger = getLogger(__name__)``` and then do the logger configuration in the app's main method/function (ie. it should never happen as a side-effect of a regular import statement).  Extra care should be taken here because this is code which is both used as a `__main__` but also imported by tests.